staging/src/k8s.io/apiserver/pkg/authorization/cel/mapper.go

```diff
@@ -31,33 +30,47 @@ type CELMapper struct {
 }
 
 // eval evaluates the given SubjectAccessReview cel mapping expression and returns a EvaluationResult.
-func (c *CELMapper) Eval(ctx context.Context, r *authorizationv1.SubjectAccessReview) ([]EvaluationResult, error) {
-	evaluations := make([]EvaluationResult, len(c.CompilationResults))
-	specVal, err := convertObjectToUnstructured(&r.Spec)
+func (c *CELMapper) Eval(ctx context.Context, r *authorizationv1.SubjectAccessReview) ([]bool, []error) {
+	specValObject, err := convertObjectToUnstructured(&r.Spec)
 	if err != nil {
-		return nil, fmt.Errorf("cel evaluation error: convert SubjectAccessReviewSpec object to unstructured failed in error: %w", err)
+		return nil, []error{fmt.Errorf("authz celmapper eval error: convert SubjectAccessReviewSpec object to unstructured failed: %w", err)}
 	}
 	va := map[string]interface{}{
-		"request": specVal.Object,
+		"request": specValObject,
 	}
+	results := make([]bool, len(c.CompilationResults))
+	evalErrors := make([]error, len(c.CompilationResults))
```

Comment:
can this simplify to `var evalErrors []error` and just append when there's an actual error, rather than allocating the whole list and having nil holes in the list?

Reply:
edit: after I typed the above comment, I realized this PR was following the admission webhook matchConditions and cel admission templates which took a similar approach. I have similar thoughts about simplifying / optimizing those.

I'd slightly prefer to simplify this code now, but if you prefer we can try to simplify all of those together in 1.30